### ISSUE TITLE
[BUG] When running on Gitlab CI storageState is not updating.

### ISSUE THREAD

**Issue by jcarn760** [microsoft/playwright#28457]
### System info
- Playwright Version: [v1.37]
- Operating System: [popOS]
- Browser: [Chromium, Firefox]
- Other info:

Followed the documentation here: https://playwright.dev/docs/auth
When running these tests locally, all performs as expected. playwright/.auth/user.json prints out the required data to authenticate and access protected routes and perform tests. in the pipeline it seems as gitlab CI pipeline it appears as if something is going wrong somewhere and it's not returning the token/header upon login and writing it to user.json. 

this is all also run inside of docker containers in the pipeline.

### Source code

playwright.config.js

```
module.exports = defineConfig({
  testDir: './playwright_tests',
  outputDir: 'playwright/test_results',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [
    [
      'html',
      { outputFolder: 'playwright_results/playwright_report', open: 'never' },
    ],
    [
      'junit',
      { outputFile: 'playwright_results/playwright_report/reports.xml' },
    ],
  ],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    baseURL: process.env.FRONTEND_URL,


    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
  },

  /* Configure projects for major browsers */
  projects: [
    // Setup project
    {
      name: 'setup', testMatch: /.*\.setup\.js/
    },
    {
      name: 'chromium',
      use: { 
        ...devices['Desktop Chrome'],
        // Use prepared auth state.
        storageState: 'playwright/.auth/user.json',
      },
      dependencies: ['setup']
    },

    {
      name: 'firefox',
      use: { 
        ...devices['Desktop Firefox'],
        // Use prepared auth state.
        storageState: 'playwright/.auth/user.json',
      },
      dependencies: ['setup']
    },
```
playwright_tests/auth.setup.js

**note** on this setup process- when running in gitlabs CI pipeline there is always a error/timeout on line 
`  await page.waitForURL('/');`

    Test timeout of 30000ms exceeded.
    Error: page.waitForURL: Navigation failed because page was closed!
    =========================== logs ===========================
    waiting for navigation to "/" until "load"
    ============================================================

when run locally this does not happen as upon a successful login it will redirect to the home '/' route.

if this line is commented out locally the tests will also fail for the same reason as they are in the pipeline

```
import { test as setup, expect } from '@playwright/test';

const authFile = 'playwright/.auth/user.json';

setup('authenticate', async ({ page }) => {
  // Perform authentication steps.
  await page.goto('/');
  await page.setViewportSize({ width: 1920, height: 1080 });
  await page.getByRole('button', { name: 'Sign in' }).click();
  await page.setViewportSize({ width: 1920, height: 1080 });
  await page.getByLabel('Username').click();
  await page.getByLabel('Username').fill('user');
  await page.getByLabel('Password', { exact: true }).click();
  await page.getByLabel('Password', { exact: true }).fill('password');
  await page.locator('form').getByRole('button', { name: 'Sign in' }).click();
  console.log((await page.context().storageState({ path: authFile })).origins[0].localStorage);
  //await page.goto("/");
  // Wait until the page receives the cookies.
  //
  // Sometimes login flow sets cookies in the process of several redirects.
  // Wait for the final URL to ensure that the cookies are actually set.

  //await expect(page).toHaveURL('/');
  await page.waitForURL('/');

  // End of authentication steps.

  await page.context().storageState({ path: authFile });
  console.log((await page.context().storageState({ path: authFile })).origins[0].localStorage);
});
```

**Steps Locally**
- Run the test with base URL http://localhost:8080
- have logs placed to check storageState playwright/.auth/user.json periodically
- log initially shows localstorage data provided in react code
- log then updates to reflect token/header localstorage data used to lift route protection
- tests perform as expected

**Steps on gitlab CI**
- Run test with base URL process.env.FRONTEND_URL
- if line  `await page.waitForURL('/');` left in then it will timeout and fail on authentication
- if that line is commented out it does not update storageState with .auth/user.json information 
- it stays the same initial information throughout tests with no token/header

**Expected**

tests to pass / obtain header/token inside user.json

**Actual**

auth setup isn't completing job successfully in gitlab CI and no token/header information is being written to be used to access protected routes in tests.

I'm assuming something is happening in auth.setup.js that isn't allowing enough time to write the user.json file before the tests or ci proceed on.

hopefully I provided enough information but would appreciate any clarification. new to playwright and automated testing.

**Comment by pavelfeldman** [comment]
When you do this:

```
const authFile = 'playwright/.auth/user.json';
```

You are loading file relative to the current working directory, which might be different on your CI. You should always resolve to absolute paths. `path.resolve(__dirname, '...')`

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
